content/200-concepts/100-components/01-prisma-schema/08-views.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: 'Views'
+metaTitle: 'How to include views in your Prisma schema'
+metaDescription: 'How to include views in your Prisma schema'
+hidePage: false
+preview: true
+tocDepth: 2
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+Support for views is currently a very early, experimental preview feature.
+
+</Admonition>
+
+<!-- TODO: add more to admonition -->
+
+Database views allow you to name and store a query. In relational databases, views are stored queries that may include columns in multiple tables. In MongoDB, views are queryable objects where the contents are defined by an [aggregation pipeline](https://www.mongodb.com/docs/manual/core/aggregation-pipeline) on other collections.
+
+This guide explains how to enable the `views` preview feature and add views to your Prisma schema.
+
+</TopBlock>
+
+## How to enable the <inlineCode>views</inlineCode> preview feature
+
+Support for views is currently in an early preview. To enable the `views` preview feature, add the `views` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema file:
+
+```prisma file=schema.prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["views"]
+}
+```
+
+## How to use views in your Prisma schema
+
+To add a view to your Prisma schema, use the `view` keyword:
+
+```prisma file=schema.prisma
+model User {
+  id      Int      @id @default(autoincrement())
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     Int    @id @default(autoincrement())
+  bio    String
+  user   User   @relation(fields: [userId], references: [id])
+  userId Int    @unique
+}
+
+// view UserInfo {
+// ... view syntax here...
+//  }
```

Comment:
I think a simple view definition could look like this:

```suggestion
view UserInfo {
  id       Int
  email String
  name String
  bio     String
}
```

And the PostgreSQL that the user would need to execute manually:

```sql
CREATE VIEW UserInfo AS
    SELECT u.id, email, name, bio
    FROM User u
    LEFT JOIN Profile p ON u.id = p.userId
```

(Both of these things should be tested of course)

(Can you review this @pimeys please?)

The two models `Profile` and `User` are technically not needed at all, but show what the existing SQL tables the view us using look like.

Reply:
Thanks - those were the fields I was imagining using, but wasn't sure from [here](https://www.notion.so/prismaio/Views-c1373f139d8b49798187186319c8c84e#1f8ae0b3820c4629a4ae26555474121a) what the syntax would end up being.

Reply:
I've added this in with an `@id` attribute on `id`, and the latest dev version seems happy with it.

In the docs the formatter currently removes the blank line between the view and the model above on saving.